Api/V8/Service/ModuleService.php

```diff
@@ -113,13 +113,31 @@ public function getRecords(GetModulesParams $params, Request $request)
         $limit = $size === BeanManager::DEFAULT_ALL_RECORDS ? BeanManager::DEFAULT_LIMIT : $size;
         $deleted = $params->getDeleted();
 
+        if (empty($fields)) {
+            $fields = array_column($bean->field_defs, 'name');
+        }
+
+        $filteredFields = array_filter(
+            $fields,
+            function ($field) use ($bean) {
+                return (
+                    in_array($field, array_column($bean->field_defs, 'name'), true)
+                    && (
+                        empty($bean->field_defs[$field]['link_type'])
+                        || $bean->field_defs[$field]['link_type'] !== 'relationship_info'
+                    )
+                );
+            }
+        );
+
```

Comment:
Potentially worth extracting this bit into its own function. The `getRecords` function is already very long.

Reply:
Even better would be to split it out the entire function into separated classes imo. Obviously not a required change though :).

Reply:
Hey hey hey, if you want me to re-write  everything just say ;) I've extracted the new logic to BeanManager class after debating a few places to put it.